lib/puppet-lint/plugins/check_unsafe_interpolations.rb

```diff
@@ -0,0 +1,32 @@
+PuppetLint.new_check(:check_unsafe_interpolations) do
+  COMMANDS = Array['command', 'onlyif', 'unless']
+  def check
+    # Gather any exec commands' resources into an array
+    exec_resources = resource_indexes.map do |resource|
+      resource if resource[:type].value == 'exec' && !(COMMANDS & resource[:param_tokens].map(&:value)).empty?
+    end
+
+    # Iterate over each command found in any exec
+    exec_resources.each do |command_resources|
+      # Filter each resource in command and check for unsafe interpolations
+      command_resources[:tokens].each do |token|
+        # Skip iteration unless token type is :NAME or :VARIABLE
+        next unless interpolation? token
```

Comment:
@chelnak I've added this for now which skips the iteration unless the token has type :NAME or :VARIABLE. One issue here is that `command` is included in the `command_resources` array and will will pass this check... 

I'm also not 100% sure what to look for in the command - should tokens of type :NAME be evaluated? Does that mean every string in the command should be evaluated?